src/pages/PullRequestPage/subroute/ComponentsSelector/ComponentsSelector.tsx

```diff
@@ -0,0 +1,89 @@
+import isUndefined from 'lodash/isUndefined'
+import { useState } from 'react'
+
+import { useLocationParams } from 'services/navigation'
+import { usePullComponents } from 'services/pull'
+import { useFlags } from 'shared/featureFlags'
+import Icon from 'ui/Icon'
+import MultiSelect from 'ui/MultiSelect'
+
+const defaultQueryParams = {
+  search: '',
+  components: [],
+}
+
+function ComponentsSelector() {
+  const { params, updateParams } = useLocationParams(defaultQueryParams)
+  const [componentSearch, setComponentSearch] = useState('')
+  const [selectedComponents, setSelectedComponents] = useState(
+    // @ts-expect-errors, useLocation params needs to be updated to have full types
+    params?.components
+  )
+  const { componentsSelect: componentsSelectFlag } = useFlags({
+    componentsSelect: false,
+  })
+  const { data, isLoading } = usePullComponents({
+    filters: {
+      components: componentSearch ? [componentSearch] : undefined,
+    },
+    options: {
+      suspense: false,
+    },
+  })
+
+  const components =
+    data?.pull?.compareWithBase.__typename === 'Comparison'
+      ? data?.pull?.compareWithBase?.componentComparisons
+      : []
+
+  if (
+    (!components?.length && !isLoading && !componentSearch) ||
+    !componentsSelectFlag
+  ) {
+    return null
+  }
+
+  const componentsNames = new Set()
```

Comment:
I think you can do 
`const componentsNames = new Set(params?.components)` and avoid the ts error

Reply:
The ts error is coming from the fact that `useLocation` params is not typed, so we would still see the error anywhere we use params :cry: